th_preprocessor/preprocess.py

```diff
@@ -267,9 +270,9 @@ def preprocess(text: str) -> str:
     text = html.unescape(text)
     text = remove_tag(text)
 
-    text = normalize_link(text)
     text = normalize_at_mention(text)
     text = normalize_email(text)
+    text = normalize_link(text)
```

Comment:
add `normalize_extension`

Reply:
change function name to `normalize_filename`